rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -536,6 +537,49 @@ void getProposedFederationAddress_whenProposedFederationExists_shouldReturnAddre
             assertEquals(expectedAddress, actualProposedFederationAddress.get());
         }
 
+        @ParameterizedTest
+        @EnumSource(FederationMember.KeyType.class)
+        void getProposedFederatorPublicKeyOfType_whenBridgeSupportReturnsEmpty_shouldReturnEmpty(FederationMember.KeyType keyType) {
+            // Arrange
+            var index = 1;
+
+            // Act
+            var actualProposedFederatorPublicKey = bridgeSupport.getProposedFederatorPublicKeyOfType(index, keyType);
+
+            // Assert
+            assertFalse(actualProposedFederatorPublicKey.isPresent());
+        }
+
+        @ParameterizedTest
+        @EnumSource(FederationMember.KeyType.class)
+        void getProposedFederatorPublicKeyOfType_whenProposedFederationExists_shouldReturnAddress(FederationMember.KeyType keyType) {
+            FederationMember member = federation.getMembers().get(0);
+            
+            // Set up public keys based on the keyType
+            byte[] expectedPublicKey;
+            switch (keyType) {
+                case BTC:
+                    expectedPublicKey = member.getBtcPublicKey().getPubKey();
+                    when(federationSupport.getProposedFederatorPublicKeyOfType(0, FederationMember.KeyType.BTC))
+                        .thenReturn(Optional.of(expectedPublicKey));
+                    break;
+                case RSK:
+                    expectedPublicKey = member.getRskPublicKey().getPubKey();
+                    when(federationSupport.getProposedFederatorPublicKeyOfType(0, FederationMember.KeyType.RSK))
+                        .thenReturn(Optional.of(expectedPublicKey));
+                    break;
+                case MST:
+                    expectedPublicKey = member.getMstPublicKey().getPubKey();
+                    when(federationSupport.getProposedFederatorPublicKeyOfType(0, FederationMember.KeyType.MST))
+                        .thenReturn(Optional.of(expectedPublicKey));
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown KeyType");
+            }
```

Comment:
Maybe this can moved out of the test into an arrange method.

Reply:
It is only used for this test, so it should be good here